### ISSUE TITLE
Issue with redirect_uri on localhost

### ISSUE THREAD

**Issue by elhamsarikhani** [MrSwitch/hello.js#326]
hi,
I'm quiet new to helloJS. I'm trying to create an app using ionic framework. 
I've tried to add a new provider for exactonline. 
I try to call the login using 
hello.init({
	exactonline: '{xxxx-xxxx-xxxxxx}'
	}, {redirect_uri: 'http://localhost:8100/callback.html'});

on local testing with 'ionic serve' I can test the login but while testing the app on my device or android emulator, as the application initiate the oauth login page on exactonline and I enter the username and password, I get the "webpage cannot be find" error message. 
I understand that "http://localhost:8100'' is not the device root for application but I don't know I should change it to what exactly.

**Comment by MrSwitch** [comment]
You are correct, the redirect_uri does not need to actually exist. Unless you want to style it - then host a page online.

What is the full URL of that page? For instance it should have some parameters including "state" and either "error" or "access_token" appended to the redirect URL like `http://localhost:8100/callback.html?state=JSONString&access_token=secret"

**Comment by elhamsarikhani** [comment]
Thank you for your fast response!
indeed the error message I am receiving is having a URL such as 
http://localhost/callback.html#token_type=bearer&access_token=XXXXX&scope=exactonline&expires_in=600&state=22&client_id=xxxxxx&network=exactonline&display=popup&callbac=_hellojs_c1&redirect_uri=http://localhost/callback.html&scope=basic

HTTP Error 404.0 - Not Found
The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.

**Comment by MrSwitch** [comment]
In the screenshot i dont see the `state` defined. It would suggest that the authenticating provider has not implemented OAuth2 correctly. It is a required field if the client has requested it https://tools.ietf.org/html/rfc6749#section-4.1.2

**Comment by elhamsarikhani** [comment]
This is their implementation guideline provided for developers:
https://developers.exactonline.com/Default.htm#OAuth Authentication.html
but if i understand you correctly, I should contact their support for this issue. correct?

**Comment by MrSwitch** [comment]
Yup please flag this error with them. please also refer them to this thread.

**Comment by elhamsarikhani** [comment]
You see I'm having the same issue with dropbox too:
![screenshot_2015-10-14-21-02-20](https://cloud.githubusercontent.com/assets/14323843/10484438/a000427a-72b7-11e5-909e-3b107c9fb501.png)

**Comment by MrSwitch** [comment]
Thats an unresolved network name error, not the same issue.
